docs/style/CODING_STYLE_GUIDE.md

```diff
@@ -0,0 +1,162 @@
+# Coding Style Guide
+
+_Revision 6_ _2024-10-28_
+
+This guide provides a small set of code guidelines used in the SDK. This guide
+reflects the currently accepted style practices for the SDK and is subject to
+change.
+
+The SDK was seeded from multiple different projects and contains contributions
+from many different companies and the SDK code therefore uses several different
+coding styles throughout the code base. Stylistically, code should attempt to
+conform to the dominant style of the code being modified, while also adhering to
+the guidelines below.
+
+## Language standard
+
+Code in the SDK conforms to the following standards. Changes to the C++ standard
+happen infrequently and there has been only one change from C++14 to C++17 over
+the course of the SDK lifetime. Changes to the Python version happen more
+frequently.
+
+| Language | Version |
+| -------- | ------- |
+| C++      | C++17   |
+| Python   | 3.10    |
+
+Product-specific software may elect to use later standards in their own work.
+
+## Coding Guidelines
+
+### Common
+
+#### When in Rome
+
+The most important convention and practice in the Matter SDK repo is "_When in
+Rome..._", per the quote below.
+
+[quote, St. Ambrose]
+
+---
+
+If you should be in Rome, live in the Roman manner; if you should be elsewhere,
+live as they do there.
+
+---
+
+Your extensions or fixes to existing code should match the prevailing of the
+original code.
+
+If you find the conventions so foreign or otherwise confusing, it may be best to
+let whoever owns the file make the necessary changes or seek the counsel of
+others in the group to find out what the right thing to do is. Never just start
+changing code wholesale for personal reasons without consulting others first.
+
+#### Commenting Out or Disabling Code
+
+Unused code shall not be disabled by commenting it out with C- or C++-style
```

Comment:
Maybe add an information what to do instead of commenting the code out, like: "simply remove it". Otherwise, one might get the impression that the unused code should not be disabled/removed at all.